libparsec/crates/types/src/manifest.rs

```diff
@@ -288,7 +295,14 @@ pub struct FileManifest {
     pub updated: DateTime,
     /// Total size of the file
     pub size: SizeInt,
-    /// Size of a single block
+    /// Recommended size of a single block.
+    ///
+    // Note blocks are not necessarily of `blocksize` size (nor aligned on `blocksize`),
+    // this is for three reasons:
+    // - Last block can be smaller than `blocksize`.
+    // - Ranges of zero-filled data in the file can be omitted.
+    // - If the total file size change significantly (e.g. file grows at lot),
+    //   `blocksize` can be updated to try to keep the number of blocks manageable.
```

Comment:
If the blocksize is updated (although not implemented at the moment), all blocks should be reshaped to match the new blocksize